### PR TITLE
Add option to skip ilgen optimizations

### PR DESCRIPTION
**Pull request by kevindean12**
Introduces a TR_DisableIlgenOpts option flag that can be set in order to skip optimizations run during ilgen.

### CODE REVIEW

File: compiler/il/OMRResolvedMethodSymbol.cpp [hunk 1]
```diff
@@ -1233,9 +1233,12 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
...
 
-            optimizer = TR::Optimizer::createOptimizer(comp, self(), true);
-            previousOptimizer = comp->getOptimizer();
-            comp->setOptimizer(optimizer);
+            if (!(comp->getOption(TR_SkipIlgenOpts)))
+               {
+               optimizer = TR::Optimizer::createOptimizer(comp, self(), true);
+               previousOptimizer = comp->getOptimizer();
+               comp->setOptimizer(optimizer);
+               }
```
Comment (0xdaryl) [inline compiler/il/OMRResolvedMethodSymbol.cpp:1241]: I was surprised to discover there isn't a means to disable this already.  I did find two unused functions `allowIlGenOptimizations()` that nothing in OMR nor any known downstream project uses, but I'd have to dig into the intent of those functions to know if, ultimately, those are the right functions to use.

In the meantime, for your case, can you rename your option `TR_DisableIlgenOpts` to be more consistent with other options for disabling features?

File: compiler/control/OMROptions.cpp [hunk 2]
```diff
@@ -1075,6 +1075,7 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
                                          TR::Options::setStaticNumericKBAdjusted, (intptr_t)&OMR::Options::_scratchSpaceLowerBound, 0, "F%d (bytes)"},
    {"searchCount=",      "O<nnn>\tcount of the max search to perform",
         TR::Options::set32BitSignedNumeric, offsetof(OMR::Options,_lastSearchCount), 0, "F%d"},
+   {"skipIlgenOpts", "O\tDo not run the optimizer during ilgen", SET_OPTION_BIT(TR_DisableIlgenOpts), "F"},
```
Comment (0xdaryl) [inline compiler/control/OMROptions.cpp:1078]: Please change the option name text here as well (and insert the new name alphabetically).

**Comment by kevindean12** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
PPC failure is known.

**Pull request merged**